src/csvcubed/writers/qbwriter.py

```diff
@@ -820,6 +982,45 @@ def _define_csvw_column_for_qb_column(
             column.structural_definition.__class__.__name__,
         )
 
+        obs_val_col: Optional[QbColumn[QbObservationValue]] = None
+        obs_val_cols = get_columns_of_dsd_type(self.cube, QbObservationValue)
+        is_single_measure = len(obs_val_cols) == 1
+
+        # If the cube is in pivoted shape, check what the column represents to set the aboutUrl
+        if self.is_cube_in_pivoted_shape:
+            # If the column represents a QbObservationValue, then simply assign the obs_val_column to this column.
+            if isinstance(column.structural_definition, QbObservationValue):
```

Comment:
Can you move the stuff inside the `if self.is_cube_in_pivoted_shape` block into a separate function? This method is getting a bit big now.

Reply:
Also, can you ensure there's some logging in these branching statements?